massa-network-worker/src/network_cmd_impl.rs

```diff
@@ -426,3 +433,37 @@ pub async fn on_ask_for_operations_cmd(
         )
         .await;
 }
+
+fn get_connection_ids(
+    worker: &mut NetworkWorker,
+    node: &NodeId,
+) -> Result<HashSet<ConnectionId>, NetworkError> {
+    let mut ids: HashSet<ConnectionId> = HashSet::new();
+    if let Some((orig_conn_id, _)) = worker.active_nodes.get(node) {
+        if let Some((orig_ip, _)) = worker.active_connections.get(orig_conn_id) {
+            worker.peer_info_db.peer_banned(orig_ip)?;
+            for (target_conn_id, (target_ip, _)) in worker.active_connections.iter() {
+                if target_ip == orig_ip {
+                    ids.insert(*target_conn_id);
+                }
+            }
+        }
+    }
+
+    Ok(ids)
+}
+
+fn get_ips(worker: &mut NetworkWorker, node: &NodeId) -> Result<Vec<IpAddr>, NetworkError> {
```

Comment:
I don't think it's possible for a node to have multiple ips. I think this should be an error case.